libs/components/src/next/form/radio/vl-radio-group.component.ts

```diff
@@ -0,0 +1,205 @@
+import { CSSResult, html, PropertyDeclarations, PropertyValues, TemplateResult } from 'lit';
+import { customElement } from 'lit/decorators.js';
+import { radioStyle } from '@domg/govflanders-style/component';
+import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
+import { vlElementsStyle } from '@domg-wc/elements';
+import radioUigStyle from './vl-radio.component.uig-css';
+import { FormControl, FormControlDefaults } from '../form-control/FormControl';
+import { VlRadioComponent } from '@domg-wc/components/next/form/radio/vl-radio.component';
+
+export const RadioGroupDefaults = {
+    ...FormControlDefaults,
+    block: false,
+    readonly: false,
+    value: '',
+};
+
+@customElement('vl-radio-group-next')
+export class VlRadioGroupComponent extends FormControl {
+    // Properties
+    private block = RadioGroupDefaults.block;
+    private readonly = RadioGroupDefaults.readonly;
+    private value: string | null = RadioGroupDefaults.value;
+
+    // Variables
+    initialValue = '';
+
+    static get styles(): (CSSResult | CSSResult[])[] {
+        return [resetStyle, baseStyle, vlElementsStyle, radioStyle, radioUigStyle];
+    }
+
+    static get properties(): PropertyDeclarations {
+        return {
+            value: { type: String, reflect: true },
+            required: { type: Boolean },
+            block: { type: Boolean },
+            readonly: { type: Boolean },
+        };
+    }
+
+    constructor() {
+        super();
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+
+        this.addEventListener('vl-checked', this.updateGroupAfterCheck);
+    }
+
+    disconnectedCallback() {
+        super.disconnectedCallback();
+
+        this.removeEventListener('vl-checked', this.updateGroupAfterCheck);
+    }
+
+    protected firstUpdated(_changedProperties: PropertyValues) {
+        super.firstUpdated(_changedProperties);
+
+        this.getRadios()?.forEach((radio) => {
+            radio.setAttribute('name', this.getAttribute('name') || '');
+        });
+
+        if (this.value) {
+            this.initialValue = this.value;
+        }
+
+        this.resetToInitialValue();
+    }
+
+    willUpdate(changedProperties: Map<string, unknown>): void {
+        super.willUpdate(changedProperties);
+
+        if (changedProperties.has('isInvalid')) {
+            this.error = this.isInvalid;
```

Comment:
Waarom is dit nodig?

Reply:
Idealiter, als het een single component was, dan kon ik gewoon afhankelijk van `isInvalid`-property error css klasses er op zetten. Hier moet gaf ik nu `error` mee aan de individuele `radio`-children. Dus als FormControlMixin `isInvalid` instelt, ga ik ook `error` op true instellen.

Maar gezien onze bespreking gisteren, dat we niet langer `error` willen resetten, is het ook niet goed dat we zelf het `error`-attribuut gaan manipuleren. Dus zal dit herwerken zodat `error` publieke API blijft en `isInvalid` interne API (maar zal dit dan wel doorgeven aan de `radio`-children.